src/artemis/devices/fast_grid_scan.py

```diff
@@ -191,7 +194,8 @@ def check_valid_and_scan():
         return st
 
     def stage(self) -> List[object]:
-        self.position_counter.put(0)
+        status = self.position_counter.set(0)
+        status.wait()
```

Comment:
you can use set_and_wait in utils.epics_pvs for this, or just add the wait onto the end of the set rather than creating the status object